lib/vehicle_factory.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'vehicle'
+
+class VehicleFactory
+  def create_vehicles(data)
+    vehicles = []
+
+    data.each do |d|
+      next if d.has_key?(:record_type) && d[:record_type] != "VEH"
+
+      d.transform_keys!(vin_1_10: :vin, model_year: :year)
+
+      case
+      when d.has_key?(:electric_vehicle_type) || d[:fuel_type] == "ELECTRIC"
+        d[:engine] = :ev
+      when d.has_key?(:fuel_type)
+        d[:engine] = d[:fuel_type].downcase.to_sym
+      else
+        puts "Unrecognized engine type: #{d[:fuel_type]}"
+      end
+
+
+      vehicles << Vehicle.new(d)
+    end
+
+    return vehicles
+  end
+end
```

Comment:
Not really much to say about this class as well, really great job. I do have a little nitpicky suggestion, and that is to be cautious when using bang methods (`!`) because of how they modify the original hash. Doesn't look like you needed the original hash here for anything important so no surprising side effects. :)